### ISSUE TITLE
Superpixel changes after update

### ISSUE THREAD

**Issue by garymlong** [ryouchinsa/Rectlabel-support#95]
The behavior of superpixel labeling has changed after the most recent update 2.62. At high superpixel size (50) the cell size is too small to make pixel labeling efficient. Previously, high superpixel size seemed to be working a lot better.

**Comment by ryouchinsa** [comment]
We removed the 960px limitation to the pixels image.
Some users are using RectLabel to edit automatically segmented mask image by their neural networks.
In this case, they want to edit the same size as the original image size, for example, 2000px x 1500px.
If you resize the image to 960px, width and height are LTE 960px, the superpixel calculation speed would be the same.

**Comment by garymlong** [comment]
My images are all large 4032 x 3024 images from iPhone 8 cameras.  At this resolution, superpixel calculation takes a long time.  @ryouchinsa thank you for the explanation about removing the 960px limitation as it explains the increased compute time.  This change was not mentioned in the release notes so it came as a big surprise.

Perhaps this should be a feature request - incorporate a scale slider into the label pixel window that will allow the user to run the superpixel calculation on scaled versions of the image. 

Superpixel is useful when large sections of the image can be grouped together.  Currently for large images, it is simply faster to use the polyline to define the boundary than it is to click on each superpixel unless I choose to resize my entire dataset prior to RectLabel.  I'd then have to upsample the output masks and worry about interpolation artifacts.

<img width="901" alt="image" src="https://user-images.githubusercontent.com/32154214/56378029-ce647b80-61d9-11e9-9aa8-24c0bc43fdf7.png">

**Comment by ryouchinsa** [comment]
Are you training 4032 x 3024 images on neural network tools?
Let us know which tool you are using.
Usually around 1000px is enough to train otherwise it takes so much time.
We talked with medical users and they decided to resize to less than 1000px.

**Comment by ryouchinsa** [comment]
> I'd then have to upsample the output masks and worry about interpolation artifacts.
This is the reason why we removed the 960px limitation.

**Comment by garymlong** [comment]
We are using the RectLabel tool to label our data for training inside Tensorflow.  We do custom tiling to process high resolution images without downsampling to improve performance against small targets.  I would prefer to not have to resize my data before going into RectLabel. 

LabelBox, Figure 8, and other companies also seem to struggle with a good implementation of Superpixel especially for large input images.  If RectLabel could incorporate multi-scale processing into the superpixel labeling it would be a huge speedup and give the performance that I think a lot of customers desire.  I love this tool, but this new change is preventing me from using superpixel labeling for my data.

**Comment by ryouchinsa** [comment]
In the previous version, there were block artifacts in mask images if your images were larger than 960px.
The new version is for users who edit mask images of large sizes.

It looks that your problem is just superpixel maximum size is too small for 4000px.
How about if you could set the maximum and minimum size of superpixel on the setting dialog?
If you use the larger superpixel size, the calculation time would be reduced.

**Comment by garymlong** [comment]
@ryouchinsa yes, I think if I could set maximum superpixel to 200 instead of 50, it would work as it did before. Since 4000px is ~4 times 960, a maximum superpixel of 200 would make sense.  The current dialog only goes to 50.

**Comment by ryouchinsa** [comment]
Through some experiments, we added "Superpixel size max" and "Superpixel image downscale factor" on the settings dialog.

For "Superpixel size max", you can change the superpixel size max from 50 to 200.
For "Superpixel image downscale factor", you can resize the image when RectLabel calculates superpixel segmentations. 
If you set to 4.0, it means the image size becomes 1 / 4. 
The calculation becomes faster, but 4x4 blocks appear in the mask image.
This is the same behavior in the previous version if your image size is around 4000px.

We would submit the new update to Apple soon.

![スクリーンショット 2019-04-20 15 13 43](https://user-images.githubusercontent.com/1954306/56453549-f3462380-637e-11e9-9013-f8317a684f97.png)

![スクリーンショット 2019-04-20 15 12 06](https://user-images.githubusercontent.com/1954306/56453551-f7724100-637e-11e9-8370-8bc28d60d558.png)

Around an ear of the dog.
![1](https://user-images.githubusercontent.com/1954306/56453556-00631280-637f-11e9-831e-765984eaae5d.jpg)

Downscale 1.0.
![2](https://user-images.githubusercontent.com/1954306/56453557-02c56c80-637f-11e9-829f-6f66422e2862.jpg)

Downscale 2.0. There are 2x2 blocks.
![3](https://user-images.githubusercontent.com/1954306/56453558-048f3000-637f-11e9-92c7-6513588c8362.jpg)

Downscale 4.0. There are 4x4 blocks.
![4](https://user-images.githubusercontent.com/1954306/56453559-0822b700-637f-11e9-92bb-8616f539e7b9.jpg)

**Comment by ryouchinsa** [comment]
We changed from "Superpixel image downscale factor" to "Superpixel image size".

If you set the value to 960, the image is resized to 960px when RectLabel calculates superpixel segmentation.
This behavior is the same as the previous version.

If you set the value to higher value, you can calculate the higher resolutional superpixels but it takes more seconds.

![スクリーンショット 2019-04-20 17 25 41](https://user-images.githubusercontent.com/1954306/56454821-76707500-6391-11e9-9aa8-7c2ffc02e798.png)

**Comment by ryouchinsa** [comment]
The new update version 2.65 was released.
Let us know your opinion.

"Superpixel size max" is to change the superpixel size max for large images. ( > 30)
"Superpixel image size" is to change the image size for superpixel calculation if the image size is larger than the size. (480px - 4096px)
https://rectlabel.com/help#draw

**Comment by ryouchinsa** [comment]
Currently If you need our support to fix this problem, please let us know.
